source/exe/main_common.cc

```diff
@@ -81,6 +84,30 @@ MainCommonBase::MainCommonBase(OptionsImpl& options) : options_(options) {
 
 MainCommonBase::~MainCommonBase() { ares_library_cleanup(); }
 
+void MainCommonBase::configureComponentLogLevels() {
+  if (!options_.componentLogLevel().empty()) {
+    std::vector<std::string> log_levels = absl::StrSplit(options_.componentLogLevel(), ',');
+    for (auto& level : log_levels) {
+      std::vector<std::string> log_name_level = absl::StrSplit(level, ':');
```

Comment:
This definitely needs more error handling. It's possible the split doesn't work, or has too many values, etc. Can you potentially factor this method out into a utility function that can be unit tested? We should also expose a sensible error message to the user (with tests) if they input an invalid string, an invalid component, etc.

Reply:
Yup. Agree. I have actually started factoring this to utility function. Will update soon.

Reply:
@mattklein123  ended up moving the parsing and validation logic to OptionsImpl so that it can be properly handled by throwing MalformedArgvException. Also added proper test cases tehre. Here we just get validated list of component and it's level which we can just use. PTAL and LMK what you think